### PR TITLE
Opens readmes and jupyter labs in browser

### PR DESCRIPTION
**Pull request by octosteve**

### CODE REVIEW

File: lib/learn_open/environments/ide_environment.rb [hunk 1]
```diff
@@ -10,6 +10,15 @@ def open_readme(lesson)
         end
       end
 
+      def open_jupyter_lab(lesson, location, editor)
+        if valid?(lesson)
+          io.puts "Opening Jupyter Lesson..."
+          run_custom_command(:browser_open, {url: lesson.to_url})
+        else
+          run_custom_command(:open_lab, {lab_name: lesson.name})
```
Comment (gj) [inline lib/learn_open/environments/ide_environment.rb:18]: Feels a bit weird to have one of the two logical paths in a method called `#open_jupyter_lab` open something other than a Jupyter lab. Would it be possible for this method to just fail to open the lab and then have the fallback lab opened in a different method?

**Review by gj — APPROVED** [review]
Nice

File: lib/learn_open/environments/ide_environment.rb (hunk 1 above)
Comment (octosteve) [reply lib/learn_open/environments/ide_environment.rb:18]: The ide is weird... if you try to run learn open for a different lab in an existing lab container it runs that command to give you a new container (And a new window in the atom download)
Really what we should do is something like 

```
valid? do
   io.puts "Opening Jupyter Lesson..."
  run_custom_command(:browser_open, {url: lesson.to_url})
end
```

So we don't keep repeating the "open a new container" logic.

What would the code for your solution look like?

**Review by talum — COMMENTED** [review]
👍

File: lib/learn_open/environments/ide_environment.rb (hunk 1 above)
Comment (gj) [reply lib/learn_open/environments/ide_environment.rb:18]: No idea! Just felt weird, but your explanation adds some necessary context

**Pull request merged**